src/server/rpc/procedures/weather/weather.js

```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
     distance *= 0.000621371;
     trace(`closest measurement is ${distance} miles from request`);
     if (distance > MAX_DISTANCE) {
-        error(`No temperature within ${MAX_DISTANCE} miles of ${lat}, ${lng}`);
+        error(`No measurement within ${MAX_DISTANCE} miles of ${lat}, ${lng}`);
     }
     return distance < MAX_DISTANCE;
 };
 
-module.exports = {
-
-    // This is very important => Otherwise it will try to instantiate this
-    isStateless: true,
-
-    // These next two functions are the same from the stateful RPC's
-    getPath: function() {
-        return '/weather';
-    },
-
-    temp: function(latitude, longitude) {
-        var url = baseUrl + '&lat=' + latitude + '&lon=' + longitude,
-            response = this.response;
-
-        trace('temp request for ' + latitude + ', ' + longitude);
-        request(url, (err, res, body) => {
-            if (err || res.statusCode < 200 || res.statusCode > 299) {
-                log('ERROR: ', (err || body));
-                return response.send('ERROR: '+(err || body));
-            }
-            body = JSON.parse(body);
 
+weather.temp = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
             var temp = 'unknown';
             if (body.main && isWithinMaxDistance(body, latitude, longitude)) {
                 temp = body.main.temp;
                 trace('Kelvin temp is '+temp+' fahrenheit is '+tuc.k2f(temp));
-                temp = Math.round(tuc.k2f(temp));
+                temp = tuc.k2f(temp).toFixed(1);
             }
-            return response.json(temp);
+            this.response.send(temp);
```

Comment:
why did you change this from `json`? This was using `json` before to make sure express didn't mistake it for a status code... I am guessing it will be ok when the number is not an integer but what if it rounds to `101.0` or something?

I think it would be safest not to change this...